### PR TITLE
DOP-2719: Generalize cta-banner icon option

### PR DESCRIPTION
**Pull request by gjchong25**
This change modifies the `cta-banner` directive to allow it to take an LG icon as an option, so that other LG icons can be used if needed. The default icon if none is specified is the "Play" icon. 

[DOP-2719 JIRA Ticket](https://jira.mongodb.org/browse/DOP-2719)

### CODE REVIEW

File: snooty/test_parser.py [hunk 1]
```diff
@@ -898,7 +898,34 @@ def test_cta_banner() -> None:
     project_config = ProjectConfig(ROOT_PATH, "", source="./")
     parser = rstparser.Parser(project_config, JSONVisitor)
 
-    # Test valid cta-banner
+    # Test valid cta-banner with specified icon
```
Comment (i80and) [inline snooty/test_parser.py:901]: My general stance is we don't need tests for just rstspec changes!

File: snooty/rstspec.toml [hunk 2]
```diff
@@ -646,8 +646,10 @@ example = """.. banner::
 [directive."mongodb:cta-banner"]
 content_type = "block"
 options.url = { type = "uri", required = true }
+options.icon = "string"
```
Comment (i80and) [inline snooty/rstspec.toml:649]: Should this be a closed set of values?

Comment (gjchong25) [reply snooty/rstspec.toml:649]: So the frontend handles the logic and error checking for dealing with whether or not the option being passed in is a valid one. If we wanted to have a closed set of values, we'd have to include all of the potential LG icons, and there are a lot.

File: snooty/test_parser.py (hunk 1 above)
Comment (gjchong25) [reply snooty/test_parser.py:901]: Oh gotcha!

File: snooty/rstspec.toml (hunk 2 above)
Comment (rayangler) [reply snooty/rstspec.toml:649]: For reference, there are about 88 potential icons to choose from [here](https://www.mongodb.design/component/icon/example/) (for now). LG does allow us to import the list of possible values on the frontend, which makes it easier to handle there with our current version of the LG Icon package if we don't want to actively maintain the same set in our rstspec.

**Review by i80and — APPROVED** [review]

**Pull request merged**